### PR TITLE
Feature/email pw login

### PR DESCRIPTION
**Pull request by ahaurw01**
Allows signing up and logging in with email and password. Shifts user creation a little bit - 

* Saves new users with their id as the given firebase `uid`.
* Abstracts the auth data object parsing a little to get us ready for multiple social login providers.
* Allows logout.

### CODE REVIEW

File: app/templates/login.hbs [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+<div class="col-md-6">
+  {{#paper-card}}
+    {{#paper-card-content}}
+      <h3>Log In</h3>
+      <form role="form" {{action "login" on="submit"}}>
+          {{paper-text classNames="login-input" type="email" label="Email" value=loginEmail}}
+          {{paper-text classNames="login-input" type="password" label="Password" value=loginPassword}}
+        {{#paper-button type="submit" classNames="md-primary" raised=true}}
+          Log In
+        {{/paper-button}}
```
Comment (SirZach) [inline app/templates/login.hbs]: should we go ahead and just provide the ability to login with any provider/email-password in one page? like adding the github login here and moving it out of the sidenav

Comment (ahaurw01) [inline app/templates/login.hbs]: Sounds good to me. I'll put that together to see what it looks like.

**Comment by ahaurw01** [comment]
Updated. You'll have to do a `bower install` if you want to test this locally. I stole some social signup styles. Also fyi I did a force push after rebasing on master.

**Comment by SirZach** [comment]
I attempted to signup with the same email address twice. The second time nothing indicated there was an error or that the email address was already in use.

**Comment by SirZach** [comment]
PR addresses #58

**Pull request merged**